src-tauri/src/main.rs

```diff
@@ -99,6 +106,156 @@ fn is_docker_running() -> bool {
     return false;
 }
 
+#[tauri::command]
+fn is_swarm_supported() -> bool {
+    match env::consts::OS {
+        "macos" => true,
+        _ => false
+    }
+}
+
+#[tauri::command]
+fn get_username() -> String {
+    let output = Command::new("whoami").output();
+
+    match output {
+        Ok(output) => {
+            output.stdout.trim().to_string()
+        },
+        Err(_) => "prem-app".to_string(),
+    }
+}
+
+#[tauri::command]
+async fn get_petals_models() -> Result<Vec<String>, String> {
+    let url = "https://health.petals.dev/api/v1/state";
+    let response = reqwest_get(url).await.map_err(|err| err.to_string())?;
+
+    if response.status().is_success() {
+        let json_data: serde_json::Value = response.json().await.map_err(|err| err.to_string())?;
+
+        let models: Vec<String> = json_data["model_reports"]
+            .as_array()
+            .unwrap_or(&vec![])
+            .iter()
+            .filter_map(|model_report| {
+                let model_info: Result<ModelInfo, _> =
+                    serde_json::from_value(model_report.clone());
+                match model_info {
+                    Ok(model_info) if model_info.state == "healthy" => Some(model_info.name),
+                    _ => None,
+                }
+            })
+            .collect();
+
+        Ok(models)
+    } else {
+        Err("Request failed".to_string())
+    }
+}
+
+
+
+#[tauri::command]
+fn is_swarm_mode_running() -> bool {
+    let output_value = get_swarm_processes();
+
+    if !output_value.is_empty() {
+        println!(
+            "🏃‍♀️ Processeses running: {}",
+            output_value.replace("\n", " ")
+        );
+        return true;
+    }
+    return false;
+}
+
+fn create_environment(handle: tauri::AppHandle) -> String {
+    // Get the application data directory
+    let app_data_dir = tauri::api::path::home_dir().expect("🙈 Failed to get app data directory");
+    let app_data_dir = app_data_dir
+        .join(".config/prem");
+    let app_data_dir = app_data_dir
+        .to_str()
+        .expect("🙈 Failed to convert app data dir path to str");
+
+    // Get create env path
+    let binding = handle.path_resolver()
+        .resolve_resource("petals")
+        .expect("🙈 Failed to find `create_env.sh`");
+    let petals_path = binding
+        .to_str()
+        .expect("🙈 Failed to convert petals path to str");
+
+    // Set env variables
+    let mut env = HashMap::new();
+    env.insert("PREM_APPDIR".to_string(), app_data_dir.to_string());
+    env.insert("PREM_PYTHON".to_string(), "prem-env".to_string());
+    env.insert("REQUIREMENTS".to_string(), format!("{petals_path}/requirements.txt"));
+
+    // Run the bash script
+    let _ = Command::new("sh")
+        .args([format!("{petals_path}/create_env.sh")])
+        .envs(env)
+        .output()
+        .expect("🙈 Failed to create env");
+    format!("{app_data_dir}/envs/prem-env/bin/python3").to_string()
+}
+
+
+#[tauri::command]
```

Comment:
as @swarnimarun recommended, looks like this is enough :)

```suggestion
#[tauri::command(async)]
```